### ISSUE TITLE
"Phishing server > listen URL" and "New Campaign > URL"?

### ISSUE THREAD

**Issue by naddika** [gophish/gophish#1979]
How does a phishing page work in terms of set up required: what comes out of the box and what's required from the admin apart from that?

Namely, do I have to *host it* elsewhere, not on the same machine where GoPhish is running, first? 

If not and if GoPhish creates it and runs it for me, then why are there 2 types of phishing addresses/URLs: 

  * one in config  --> "phishing server/listen URL"
  * the 2nd one is in "new campaign" --> URL
?

* How are these different from each other?

* For "new campaign > URL", how will actually GoPhish be able to capture the credentials from that page/URL if that page isn't run by GoPhish? And GoPhish and that page may even run on different machines.

**Comment by glennzw** [comment]
You run the gophish binary and by default it will listen on port 443 (admin console) and 80 (landing pages).

When you create a campaign, you need to enter the URL/domain that will route to the above port 80. The reason it's setup like this is that you may want to create proxies in front of the landing page. Or perhaps you want a domain, not an IP, or a different domain for each campaign. Whatever you enter for the listening URL, it needs to end up being routed to the above mentioned port 80 (or no statistics will be gathered).

https://docs.getgophish.com/user-guide/documentation/campaigns

**Comment by naddika** [comment]
@glennzw I've seen the documentation, but it's written in a brief manner.

The landing page itself will be hosted by GoPhish at "phishing server/listen URL" on the port 80. Which is always on *same server* where Gophish runs because it's also runs by Gophish. A user will enter his credentials there. Correct?

>When you create a campaign, you need to enter the URL/domain that will route to the above port 80. 
But that's optional? I can, theoretically, ask a user go to "phishing server/listen URL"  in an email directly.

>Whatever you enter for the listening URL, it needs to end up being routed to the above mentioned port 80 (or no statistics will be gathered).

In what sense "being routed"? By whom, from where? Where at? 

And why? What if I ran "phishing server/listen URL"  at the port 8081?

**Comment by glennzw** [comment]
> The landing page itself will be hosted by GoPhish at "phishing server/listen URL" on the port 80 (customizable). Which is always on _same server_ where Gophish runs because it's also run by Gophish. A user will enter his credentials there.
> 
> Correct?

Yes.

> 
> > When you create a campaign, you need to enter the URL/domain that will route to the above port 80.
> 
> But that's optional? I can, theoretically, ask a user go to "phishing server/listen URL" directly, in an email .
> After all, a user will still be redirected to a landing page which runs on bare IP address, unless I buy a domain for it.

It's not optional. The value you enter here will be a place holder for the {{.URL}} variable you enter in an email. e.g if your listener is entered as http://1.1.1.1 and in the email you have "Please click {{.URL}}", the result in the user's inbox will be "Please click http://1.1.1.1?rid=abc1234" (where rid=abc1234 is a random unique ID to track that user have clicked the link).
 
> > Whatever you enter for the listening URL, it needs to end up being routed to the above mentioned port 80 (or no statistics will be gathered).
> 
> In what sense "being routed"? By whom, from where? Where at?

Routed by you - for example via nginx or Caddy. If you don't know what these things are you can ignore this.
 
> And why? What if I ran "phishing server/listen URL" at the port 8081?

You can run it on any port you like. I run my landing pages on 443.

Hope that helps.

**Comment by naddika** [comment]
Thanks.

Is a proxy or intermediate server-domain ("new campaign > URL") supposed to be the one that

a) has a legit, semi trustworthy domain name? Such as: payplal.com

b) redirects a user to the phishing/landing page, which is controlled by GoPhish, **right away** and without any action required from a user

c) may not have any html page at all

?

**Comment by naddika** [comment]
> It's not optional. The value you enter here will be a place holder for the {{.URL}} variable you enter in an email. e.g if your listener is entered as http://1.1.1.1 and in the email you have "Please click {{.URL}}", the result in the user's inbox will be "Please click http://1.1.1.1?rid=abc1234" (where rid=abc1234 is a random unique ID to track that user have clicked the link).

Why the "listener"? The listener is Gophish phishing server. No?
But in "new campaign > URL" this is an URL of **a proxy** which will redirect a user to that listener.

And if the {{.URL}} is the URL of the Gophish phishing server/listener, then what variable holds the URL of a proxy specified at "new campaign > URL"?

**Comment by glennzw** [comment]
>> Why the "listener"? The listener is Gophish phishing server. No?

The gophish service binds to whatever you specify in your config.json (e.g. 1.1.1.1). Perhaps you don't want http://1.1.1.1 in your phishing email, as that looks suspicious. You want `http://legitdomain.com`. So, you can go register legitdomain.com and give it a DNS record of 1.1.1.1. In your gophish listner field you may then enter http://legitdomain.com. This will result in email campaigns having URLs of the form "Hey, click this link: http://legitdomain.com/?rid=123abc4"

Please disregard the proxy comment, it's not relevant.

**Comment by naddika** [comment]
> Please disregard the proxy comment, it's not relevant.

In that case, the "new campaign > **URL**", **is** optional?  And I can simply use http://legitdomain.com in all my emails to send all users straigt to the phishing page? Assuming that my server with Gophish has the domain name legitdomain.com

**Comment by glennzw** [comment]
It's not optional. If you use http://legitdomain.com in your emails gophish will have no way to track who clicked the link.

When you enter http://legitdomain.com in the URL parameter, gophish uses this variable in your phishing campaigns replacing the placeholder {{.URL}} with `https://legitdomain.com` + `/rid?=<ridForThatUser>` resulting in, for example, `https://legitdomain.com/rid?=abc1234`. This allows gophish to track who clicked the link.

**Comment by naddika** [comment]
Thanks.

**Comment by glennzw** [comment]
No worries, good luck!

**Comment by naddika** [comment]
I'm still confused.

* I don't need to track anyone per campaign at this time, I just to make it as simple as possible. Should I use the phishing host for "new campaign URL"  --> mydomain123.com?

* "URL of **Gophish** Listener" -- what's the listener? Does Gophish spawn a listener as well as the phishing server/page?

**Comment by glennzw** [comment]
To make it as simple as possible you're best off using the URL parameter.

If not, there's no way to load a landing page via an email as the landing pages are loaded per rid- otherwise there would be no way to track which user submitted data.

If you access the gophish server on 80 directly you'll get a 404, you need to supply the rid parameter of the user of the campaign. This is done for you via the {{.URL}} placeholder (so the user is sent to foo.com/?rid=abc1234 and the appropriate landing page loads). Otherwise there's no way for Gophish to know which landing page you want to load.

If you really don't want to use the URL parameter you'd need to host your own landing page and capture submissions yourself.

**Comment by naddika** [comment]
> This is done for you via the {{.URL}} placeholder (so the user is sent to foo.com/?rid=abc1234 and the appropriate landing page loads). 

But at "new campaign > URL" it's a whole URL, not just a parameter. In the documentation it says

> This is the URL that populates the {{.URL}} template value, commonly used in email templates. This should be a URL or IP address that points to the Gophish phishing server and is reachable by the recipient.


Suppose I had some domain "my-domain1.com". I specify that in "new campaign > URL". So, what'll happen next? User will arrive there, via  {{.URL}} in an email, and that'll be it. Gopshish has no access to that domain whatsoever.


Nowhere does it mention whether

* it has or not to belong to **the same server** where Gopshih runs or not
* it has or not contain "rid" parameter
* it has then to redirect to a user "my-gophish-phishing-page.com?rid=<original_campaign_parameter" where Gopshish runs

**Comment by glennzw** [comment]
The RID parameter is out of your control, and you don't need to worry about it.

Whatever text you enter in that URL box will be replaced in campaigns emails with where you place {{.URL}}.

e.g :
- Create Email Template with text "Please click {{.URL}}"
- Specify URL of "http://192.168.0.1" in your Campaign 

Now a user will receive an email with "Please click http://192.168.0.1/?rid=aHsbzn13"

When a user browses to that address, the gophish landing page you specified when creating the campaign will be loaded.

The only requirent is that what you enter in URL is reachable by your target users. In the above example that locally reputable IP probably isn't, that's why you need to run gophish on an Internet accessible server.

If you choose to use a domain you just add an A record to point from the chosen domain to the IP address on which gophish runs.

**Comment by naddika** [comment]
Ok, but still incompletely. No answer to what I've asked several times already. 

> Now a user will receive an email with "Please click http://192.168.0.1/?rid=aHsbzn13"
> When a user browses to that address, **the gophish landing page** you specified when creating the campaign will be loaded.

Why would a landing page be loaded if that address, 192.168.0.1, may not even belong to a server where Gopshish runs? 


> The only requirent is that what you enter in URL is reachable by your target users.

That is, I may specify "google.com" at "new campaign > URL". And everything would work because it meets the only requirement of "being accessible my target user".

**Comment by S0larflare** [comment]
I think you’ve fundementally misunderstood something. The campaign url should (must) point to the box that is hosting GoPhish.

So if your box is at 202.111.200.1 then put that. If it also has an associated domain of “http://totally not phishing.com then you can use that too

Sent from my iPhone

On 25 Sep 2020, at 18:59, Naddika <notifications@github.com> wrote:

﻿

Ok, but still incompletely. No answer to what I've asked several times already.

Now a user will receive an email with "Please click http://192.168.0.1/?rid=aHsbzn13"
When a user browses to that address, the gophish landing page you specified when creating the campaign will be loaded.

Why would a landing page be loaded if that address, 192.168.0.1, may not even belong to a server where Gopshish runs?

The only requirent is that what you enter in URL is reachable by your target users.

That is, I may specify "google.com" at "new campaign > URL". And everything would work because it meets the only requirement of "being accessible my target user".

**Comment by jordan-wright** [comment]
As @S0larflare mentioned, it seems there is some misunderstandings about how Gophish manages landing pages.

To start, in the config.json you're just telling Gophish how to listen for new connections. Usually it's as simple as setting this to `0.0.0.0:80` and calling it a day which tells Gophish to listen on all interfaces. This means that requests to the Gophish server's IP address or any domain name that points to the Gophish server will be handled by Gophish itself.

Great, so the request is received. Why is that important? To answer this, we need to talk about landing pages.

When you make a campaign, two pieces of information you provide are a landing page and a URL. It is expected that this URL points to your Gophish server. Then, when Gophish creates emails and sends them to recipients, it will generate a random string and add it to the URL. It looks like this `?rid=1234abc`.

So now a recipient gets the email and clicks on the link, which points to Gophish because that's the URL we used in a campaign. In this case, Gophish sees the `rid` parameter, looks up which campaign/recipient this belongs to, figures out which landing page to return, and sends that HTML to the recipient.

In a nutshell, everything is self-contained in Gophish. You don't need to host phishing pages anywhere else. You just need to set the URL in a campaign to point to Gophish and you'll be all set.

**Comment by naddika** [comment]
> I think you’ve fundementally misunderstood something. The campaign url should (must) point to the box that is hosting GoPhish.

Right.  But never that has been mentioned in this topic. All that's been mentioned has been "**any URL** to which a recipient has access to". Google, github, yahoo and million other websites...... unless they're blocked in a country where a recipient/user lives, I could enter one of those URLs and it'd work and magically redirect a user to my phishing server.

Nor the documention mentions that either. **URL to which a user has access to.**


How does that URL have to  relate to a machine where Gophish run at, in order for the whole scheme to work? Go figure.

**Comment by naddika** [comment]
Now it's become more or less clear, thanks.

**Comment by glennzw** [comment]
> Now it's become more or less clear, thanks.

Great, I'm glad the others could clear it up. Happy phishing.

**Comment by MrJeffyw** [comment]
hi guys, what if i set the "URL" as "http://mydomain.com/xxx/xxx.jsp" not just "http://mydomain.com"?
